test/widgets/inbox_test.dart

```diff
@@ -310,6 +352,29 @@ void main() {
           checkAppearsUncollapsed(tester, findSectionContent);
         });
 
+        testWidgets('collapse all-DMs section after scroll', (tester) async {
+          await setupVarious(tester);
+
+          final listFinder = find.byType(Scrollable);
+          final dmFinder = find.text(eg.otherUser.fullName).hitTestable();
+
+          // Scroll part of [_AllDmsSection] offscreen.
+          await dragUntilInvisible(
+            tester, dmFinder, listFinder, const Offset(0, -50));
```

Comment:
Still used the variables instead of directly populating the `finder`s for better readability, but please let me know if you think the other way is better!

Reply:
For this `dragUntilInvisible` call, it can go either way — there's some value in that variable name `dmFinder` in particular. So arranging it this way is fine.

For `headerRow` below, though, definitely better to inline. When the definition reads:
```dart
          final headerRow = findAllDmsHeaderRow(tester);
```
there's not really anything the variable name is expressing that wouldn't also be clear from seeing the expression `findAllDmsHeaderRow(tester)` directly. Similarly `headerRowAfterTap`.